src/components/StaticWebsites.ts

```diff
@@ -280,11 +280,10 @@ class StaticWebsiteConstruct extends ComponentConstruct {
         }
 
         log(`Uploading directory '${this.configuration.path}' to bucket '${bucketName}'`);
-        // TODO proper upload, without going through a subcommand
-        spawnSync("aws", ["s3", "sync", "--delete", this.configuration.path, `s3://${bucketName}`], {
-            stdio: "inherit",
-        });
-        await this.clearCDNCache();
+        const changes = await s3Sync(this.serverless.getProvider("aws"), this.configuration.path, bucketName);
```

Comment:
en règle générale en ts, on préfère passer un objet d'arg et un objet de retour pour bénéficier des tooltips du typage. ton arg de retour, si c'est un boolean, je partirai sur un `hasChanges` plutôt que juste `changes`

```suggestion
        const { hasChanges } = await s3Sync({
          provider: this.serverless.getProvider("aws"),
          localPath: this.configuration.path,
          bucketName
        });
```